sdk/data_cosmos/src/operations/create_collection.rs

```diff
@@ -2,52 +2,87 @@ use crate::headers::from_headers::*;
 use chrono::{DateTime, Utc};
 
 #[derive(Debug, Clone)]
-pub struct CreateCollectionOptions {
+pub struct CreateCollectionBuilder {
+    client: DatabaseClient,
     partition_key: PartitionKey,
     consistency_level: Option<ConsistencyLevel>,
     indexing_policy: Option<IndexingPolicy>,
+    collection_name: String,
     offer: Option<Offer>,
+    context: Context,
 }
 
-impl CreateCollectionOptions {
-    pub fn new<P: Into<PartitionKey>>(partition_key: P) -> Self {
+impl CreateCollectionBuilder {
+    pub(crate) fn new(
+        client: DatabaseClient,
+        collection_name: String,
+        partition_key: PartitionKey,
+    ) -> Self {
         Self {
-            partition_key: partition_key.into(),
+            client,
+            collection_name,
+            partition_key,
             consistency_level: None,
             indexing_policy: None,
             offer: None,
+            context: Context::new(),
         }
     }
 
     setters! {
         consistency_level: ConsistencyLevel => Some(consistency_level),
         indexing_policy: IndexingPolicy => Some(indexing_policy),
         offer: Offer => Some(offer),
+        context: Context => context,
     }
 
-    pub(crate) fn decorate_request(
-        &self,
-        request: &mut HttpRequest,
-        collection_name: &str,
-    ) -> crate::Result<()> {
-        azure_core::headers::add_optional_header2(&self.offer, request)?;
-        azure_core::headers::add_optional_header2(&self.consistency_level, request)?;
+    pub fn into_future(self) -> CreateCollection {
+        Box::pin(async move {
+            let mut request = self.client.cosmos_client().prepare_request_pipeline(
+                &format!("dbs/{}/colls", self.client.database_name()),
+                http::Method::POST,
+            );
+            azure_core::headers::add_optional_header2(&self.offer, &mut request)?;
+            azure_core::headers::add_optional_header2(&self.consistency_level, &mut request)?;
 
-        let collection = CreateCollectionBody {
-            id: collection_name,
-            indexing_policy: &self.indexing_policy,
-            partition_key: &self.partition_key,
-        };
+            let collection = CreateCollectionBody {
+                id: &self.collection_name,
+                indexing_policy: &self.indexing_policy,
+                partition_key: &self.partition_key,
+            };
 
-        request.set_body(bytes::Bytes::from(serde_json::to_string(&collection)?).into());
-        Ok(())
+            request.set_body(bytes::Bytes::from(serde_json::to_string(&collection)?).into());
+
+            let response = self
+                .client
+                .pipeline()
+                .send(
+                    self.context.clone().insert(ResourceType::Collections),
+                    &mut request,
+                )
+                .await?;
+
+            Ok(CreateCollectionResponse::try_from(response).await?)
```

Comment:
Excluding formatting & straight forwards style issues highlighted by clippy (such as the following), LGTM.

```suggestion
           CreateCollectionResponse::try_from(response).await
```